jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdAuthenticator.java

```diff
@@ -0,0 +1,481 @@
+//      http://www.opensource.org/licenses/apache2.0.php
+//
+//  You may elect to redistribute this code under either of these licenses.
+//  ========================================================================
+//
+
+package org.eclipse.jetty.security.openid;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.security.SecureRandom;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpVersion;
+import org.eclipse.jetty.http.MimeTypes;
+import org.eclipse.jetty.security.LoginService;
+import org.eclipse.jetty.security.ServerAuthException;
+import org.eclipse.jetty.security.UserAuthentication;
+import org.eclipse.jetty.security.authentication.DeferredAuthentication;
+import org.eclipse.jetty.security.authentication.LoginAuthenticator;
+import org.eclipse.jetty.security.authentication.SessionAuthentication;
+import org.eclipse.jetty.server.Authentication;
+import org.eclipse.jetty.server.Authentication.User;
+import org.eclipse.jetty.server.Request;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.server.UserIdentity;
+import org.eclipse.jetty.util.MultiMap;
+import org.eclipse.jetty.util.URIUtil;
+import org.eclipse.jetty.util.log.Log;
+import org.eclipse.jetty.util.log.Logger;
+import org.eclipse.jetty.util.security.Constraint;
+
+/**
+ * OpenId Connect Authenticator.
+ *
+ * <p>This authenticator implements authentication using OpenId Connect on top of OAuth 2.0.
+ *
+ * <p>The authenticator redirects unauthenticated requests to the identity providers authorization endpoint
+ * which will eventually redirect back to the redirectUri with an authorization code which will be exchanged with
+ * the token_endpoint for an id_token. The request is then restored back to the original uri requested.
+ * {@link SessionAuthentication} is then used to wrap Authentication results so that they are associated with the session.</p>
+ */
+public class OpenIdAuthenticator extends LoginAuthenticator
+{
+    private static final Logger LOG = Log.getLogger(OpenIdAuthenticator.class);
+
+    public static final String __CLAIMS = "org.eclipse.jetty.security.openid.claims";
+    public static final String __RESPONSE = "org.eclipse.jetty.security.openid.response";
+    public static final String __ERROR_PAGE = "org.eclipse.jetty.security.openid.error_page";
+    public static final String __J_URI = "org.eclipse.jetty.security.openid.URI";
+    public static final String __J_POST = "org.eclipse.jetty.security.openid.POST";
+    public static final String __J_METHOD = "org.eclipse.jetty.security.openid.METHOD";
+    public static final String __CSRF_TOKEN = "org.eclipse.jetty.security.openid.csrf_token";
+    public static final String __J_SECURITY_CHECK = "/j_security_check";
```

Comment:
Get rid of the `__` in front of `public` constants.
Why are these `public`?

Reply:
I have removed the `__` parts. 
This was based off the `FormAuthenticator` where all the constants are prefixed with `__` and are `public`.